contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+
+/**
+ * @title Art Blocks Dependency Registry, V0.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * Permissions managed by ACL contract
+ */
+contract DependencyRegistryV0 is Ownable {
+    using BytecodeStorage for string;
+    using BytecodeStorage for address;
+    using Bytes32Strings for bytes32;
+    using Strings for uint256;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    uint8 constant AT_CHARACTER_CODE = uint8(bytes1("@")); // 0x40
+
+    /// admin ACL contract
+    IAdminACLV0 public adminACLContract;
+
+    event ProjectDependencyOverrideAdded(
+        address indexed _coreContractAddress,
+        uint256 indexed _projectId,
+        bytes32 _dependencyTypeId
+    );
+
+    event ProjectDependencyOverrideRemoved(
+        address indexed _coreContractAddress,
+        uint256 indexed _projectId
+    );
+
+    event DependencyTypeAdded(
+        bytes32 indexed _dependencyTypeId,
+        string _preferredCDN,
+        string _preferredRepository,
+        string _projectWebsite
+    );
+
+    event DependencyTypeRemoved(bytes32 indexed _dependencyTypeId);
+
+    event DependencyTypeProjectWebsiteUpdated(
+        bytes32 indexed _dependencyTypeId,
+        string _projectWebsite
+    );
+
+    event DependencyTypePreferredCDNUpdated(
+        bytes32 indexed _dependencyTypeId,
+        string _preferredCDN
+    );
+
+    event DependencyTypePreferredRepositoryUpdated(
+        bytes32 indexed _dependencyTypeId,
+        string _preferredRepository
+    );
+
+    event DependencyTypeAdditionalCDNUpdated(
+        bytes32 indexed _dependencyTypeId,
+        string _additionalCDN,
+        uint256 _additionalCDNIndex
+    );
+
+    event DependencyTypeAdditionalCDNRemoved(
+        bytes32 indexed _dependencyTypeId,
+        uint256 indexed _additionalCDNIndex
+    );
+
+    event DependencyTypeAdditionalRepositoryUpdated(
+        bytes32 indexed _dependencyTypeId,
+        string _additionalRepository,
+        uint256 _additionalRepositoryIndex
+    );
+
+    event DependencyTypeAdditionalRepositoryRemoved(
+        bytes32 indexed _dependencyTypeId,
+        uint256 indexed _additionalRepositoryIndex
+    );
+
+    event DependencyTypeScriptUpdated(bytes32 indexed _dependencyTypeId);
+
+    struct DependencyType {
+        string preferredCDN;
+        mapping(uint256 => string) additionalCDNs;
+        uint24 additionalCDNCount;
+        string preferredRepository;
+        mapping(uint256 => string) additionalRepositories;
+        uint24 additionalRepositoryCount;
+        string projectWebsite;
+        uint24 scriptCount;
+        // mapping from script index to address storing script in bytecode
+        mapping(uint256 => address) scriptBytecodeAddresses;
+    }
+
+    EnumerableSet.Bytes32Set private _dependencyTypes;
+    mapping(bytes32 => DependencyType) dependencyTypeInfo;
+    mapping(address => mapping(uint256 => bytes32)) projectDependencyOverrides;
```

Comment:
WDYT about a dependency-registry level, admin-configurable, enumerable set of supported core contract `address`es? That would restrict us to, for example, only define overrides for our flagship core contracts. Would that be a good thing?

Reply:
Quite like this idea a lot–this makes sense to me.